docs/api-guide.md

```diff
@@ -0,0 +1,718 @@
+# Java API Guide
+
+This guide briefly explains how to use Scalar DB Java API.
```

Comment:
I think it's helpful to have indexes like below.
```
* [Administrative operations](link)
* [Transactional operations](link)
```

Reply:
@brfrn169 Thank you for the update. But, the link seems not working.

Reply:
@feeblefakie Sorry for that... 😞 I fixed the broken links. Please take a look again when you have time!